pkg/vm/stack.go

```diff
@@ -163,64 +128,43 @@ func NewStack(n string) *Stack {
 
 func newStack(n string, refc *refCounter) *Stack {
 	s := new(Stack)
+	s.elems = make([]Element, 0, 16) // Most of uses are expected to fit into 16 elements.
 	initStack(s, n, refc)
 	return s
 }
 func initStack(s *Stack, n string, refc *refCounter) {
 	s.name = n
 	s.refs = refc
-	s.top.next = &s.top
-	s.top.prev = &s.top
-	s.len = 0
+	s.Clear()
 }
 
 // Clear clears all elements on the stack and set its length to 0.
 func (s *Stack) Clear() {
-	s.top.next = &s.top
-	s.top.prev = &s.top
-	s.len = 0
+	if s.elems != nil {
```

Comment:
By the way, we don't clear the reference counter. This is not correct https://github.com/neo-project/neo-vm/blob/master/src/neo-vm/EvaluationStack.cs#L38, and this is used in the CLEAR opcode.

Reply:
Fixed.